src/esw/cameras/cameras.py

```diff
@@ -0,0 +1,281 @@
+            # if the video source does not exist
+            return False
+
+    def get_device_number(self) -> int:
+        """Returns the camera device which the pipeline is assigned"""
+        return self.device_number
+
+    def is_currently_streaming(self) -> bool:
+        """Returns whether or not the pipeline is
+        assigned an active camera device"""
+        return self.device_number != -1
+
+    def update_device_number(
+            self, arguments: list[str], index: int) -> None:
+        """Takes in a requested camera device.
+        The pipeline will then be assigned this camera device
+        unless it does not exist. A -1 means that the pipeline
+        is no longer assigned a device."""
+        self.device_number = index
+        if index != -1:
+            self.video_source = self.video_source[index]
+            if self.video_source is not None:
+                self.video_output = jetson.utils.videoOutput(
+                    f"rtp://{self.current_ip}",
+                    argv=arguments)
+            else:
+                print(f"Unable to play camera {index} on \
+                    {self.current_ip}.")
+                self.device_number = -1
+        else:
+            self.video_source = None
+
+    def update_video_output(
+            self, arguments: list[str],
+            current_mission_ip: str) -> None:
+        """Updates the video output to ensure that pipeline
+        is streaming to the proper IP"""
+        self.current_ip = current_mission_ip
+        self.video_output = jetson.utils.videoOutput(
+                f"rtp://{self.current_ip}",
+                argv=arguments)
+
+
+class PipelineManager:
+    def __init__(self) -> None:
+        self.resolution_arguments_map: dict[str, list[str]] = {}
+        for input_map in rospy.get_param("cameras/input").items():
+            quality = input_map['resolution']
+            self.resolution_arguments_map[quality] = \
+                input_map['arguments']
+
+        self.default_mission = rospy.get_param("cameras/default_mission")
+        self.max_video_device_id_number = \
+            rospy.get_param("cameras/max_video_device_id_number")
+
+        self.mission_ips_map = {}
+        self.mission_resolution_map = {}
+        for missions_map in rospy.get_param("cameras/missions").items():
+            mission_name = missions_map['name']
+            self.mission_ips_map[mission_name] = missions_map['ips']
+            self.mission_resolution_map[mission_name] = \
+                missions_map['resolution']
+
+        self.current_mission = self.default_mission
+
+        self.arguments: list[str] = self.resolution_arguments_map[
+            self.mission_resolution_map[self.current_mission]['resolution']]
+        self.video_sources = [None] * self.max_video_device_id_number
+
+        number_of_pipelines = \
+            rospy.get_param("cameras/number_of_pipelines")
+        self.pipelines: list[Pipeline] = \

```

Comment:
Does it make sense to store pipelines as a dictionary rather than an array? We would have to think about what the key would be - possibly an enum?

Reply:
Not sure why we would make it a dictionary